src/components/TodoItem.tsx

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useContext,
+  useState,
+  useEffect,
+  useRef,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { TodosContext } from './TodosContext';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { id, title, completed } = todo;
+  const { todos, setTodos } = useContext(TodosContext);
+
```

Comment:
You are using the `todos` and `setTodos` from context, but it may cause unnecessary re-renderings of this component every time any of the todos changes. You should refactor your code to use specific actions to handle the todos operations like complete a todo, delete a todo, update a todo.